### DISCUSSION TITLE
Specified provider "DmlExecution Provider"

### DISCUSSION THREAD

**Discussion by Ruuto6299** [azuritecoin/OnnxDiffusersUI#58] [Q&A]
So I am a bit of an noob at Stable diffusion and I got myself a RX 6600 to try this program out. I was able to install everything properly but it's not detecting my GPU and it's giving me this error, having a bit of help would really help out a lot

![image](https://user-images.githubusercontent.com/120725609/208097461-aa2d0fb9-03f5-4916-a2c3-f0ae62744080.png)

**Comment by azuritecoin** [comment]
I suggest re-installing `onnxruntime-directml` package. First activate the virtual environment:  
`.\virtualenv\Scripts\activate.bat`

Then use pip to re-install the package:  
`pip install --ignore-installed onnxruntime-directml`

**Comment by Ruuto6299** [comment]
I tried it out but unfortunately didn't fixed it, still don't have the execution provider nor using my GPU

is it possible because of this error? on setup.bat it only installs protobuf 3.02.01 and updating it makes it 4.2.12
![image](https://user-images.githubusercontent.com/120725609/208710740-25dff953-6f41-4387-a12c-afa5a57e50f1.png)

**Reply by azuritecoin** [reply]
That error can be ignored for now... 

I messed up in the last message. To re-install it should be this:  
`pip uninstall onnxruntime-directml && pip install --ignore-installed onnxruntime-directml`

**Comment by Ruuto6299** [comment]
I tried but still no dice

is it because when trying to give a token it's giving me this error?
![image](https://user-images.githubusercontent.com/120725609/211567951-20a46f15-5956-4ce2-9480-0d151f60f31d.png)

**Reply by azuritecoin** [reply]
The error says you entered a invalid token. There's some info in my guide here: https://github.com/azuritecoin/OnnxDiffusersUI#logging-in-using-your-token

**Comment by Ruuto6299** [comment]
this time I retried it now with the Token passed, but unfortunately it still isn't working. Starting to feel a bit sad now since I just really can't make the program to use my GPU..
![image](https://user-images.githubusercontent.com/120725609/213236052-28434f7f-111a-4793-8165-da6b318621b9.png)

**Reply by azuritecoin** [reply]
Well it's running on CPU. Try this (make sure you've activated the virtual environment):  
```shell
pip uninstall onnxruntime-directml
pip install pip install -U "protobuf<=3.20.1" onnxruntime-directml
```

Then run it again.

**Comment by Ruuto6299** [comment]
I FINALLY GOT IT TO WORK!
I'm dumb as a brick

it now worked when I tried using the programs and whatnot while virtual environment is still on
